File: src/main/java/net/atos/zac/app/informatieobjecten/converter/RestOndertekeningConverter.java
/*
 * SPDX-FileCopyrightText: 2022 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.zac.app.informatieobjecten.converter;

import net.atos.zac.app.informatieobjecten.model.RESTOndertekening;
import nl.info.client.zgw.drc.model.generated.Ondertekening;

public class RestOndertekeningConverter {
    public static RESTOndertekening convert(final Ondertekening ondertekening) {
        final RESTOndertekening restOndertekening = new RESTOndertekening();
        restOndertekening.soort = ondertekening.getSoort().name().toLowerCase();
        restOndertekening.datum = ondertekening.getDatum();
        return restOndertekening;
    }
}

File: src/main/java/net/atos/zac/app/informatieobjecten/converter/RestOndertekeningConverter.kt
/*
 * SPDX-FileCopyrightText: 2022 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.app.informatieobjecten.converter

import net.atos.zac.app.informatieobjecten.model.RestOndertekening
import nl.info.client.zgw.drc.model.generated.Ondertekening
import java.util.Locale

fun Ondertekening.toRestOndertekening() = RestOndertekening(
    soort = this@toRestOndertekening.getSoort().name.lowercase(Locale.getDefault()),
    datum = this@toRestOndertekening.getDatum()
)
